test/specs/languages/test-emacs-lisp.el

```diff
@@ -0,0 +1,57 @@
+;;; test-emacs-lisp.el --- Flycheck Specs: Haskell      -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2013-2016 Sebastian Wiesner and Flycheck contributors
+
+;; Author: Clément Pit-Claudel <clement.pitclaudel@live.com>
+
+;; This file is not part of GNU Emacs.
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;; Specs for Emacs Lisp support.
+
+;;; Code:
+
+(require 'flycheck-buttercup)
+
+(defconst tel--excluded-checkdoc-vars
+  '(checkdoc-autofix-flag checkdoc-bouncy-flag checkdoc-minor-mode-string)
+  "Checkdoc variables that should not be passed to the checker.")
+
+(defun tel--custom-var-p (var)
+  "Check if VAR is a customization pair describing a variable."
+  (eq (cadr var) 'custom-variable))
+
+(describe "Language Emacs Lisp"
+
+  (describe "Checkdoc"
+    (it "exports the right customizable variables"
+      (require 'checkdoc)
+      (let* ((checkdoc-custom-group
+              (get 'checkdoc 'custom-group))
+             (checkdoc-custom-pairs
+              (seq-filter #'tel--custom-var-p checkdoc-custom-group))
+             (checkdoc-custom-vars
+              (seq-map #'car checkdoc-custom-pairs))
+             (checkdoc-relevant-vars
+              (cl-set-difference checkdoc-custom-vars tel--excluded-checkdoc-vars)))
```

Comment:
Please use `seq-difference` instead of `cl-set-difference` here.  Just my personal preference for `seq` over `cl-lib` :relaxed: